### DISCUSSION TITLE
How to update my fork in GitHub Desktop

### DISCUSSION THREAD

**Discussion by danielniccoli** [community/community#24064] [Apps, API and Webhooks]
<p>I started to contribute to a project. Using GitHub desktop, I forked the project and started working on it. The client keeps the code on my computer synced with the fork in my repositories.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/8/8/8888d1ac8fdf682918097548b2afdd7814780970.png" data-download-href="/uploads/short-url/jtQ4iQX34eWYYyUDa672BIf7pra.png?dl=1" title="grafik" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/2971735/181081780-befccc2d-16d8-419e-a487-19a5b963fb8e.png" alt="grafik" data-base62-sha1="jtQ4iQX34eWYYyUDa672BIf7pra" width="690" height="102" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/8/8/8888d1ac8fdf682918097548b2afdd7814780970_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">grafik</span><span class="informations">926×137 7.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>However, when I visit the fork-repo on <a href="http://github.com" rel="noopener nofollow ugc">github.com</a>, the page tells me that it is a couple of commits behind upstream. Why is this not shown in the GitHub desktop client?</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/5/b/5b402f757d9f3ee96697e09f23a63b57e1e823e1.png" data-download-href="/uploads/short-url/d1eXUInfPZNyZWnR5foaq3cGoJb.png?dl=1" title="grafik" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/2971735/181081781-20c20c27-6f24-4a2b-a8c6-438898cc0d2b.png" alt="grafik" data-base62-sha1="d1eXUInfPZNyZWnR5foaq3cGoJb" width="690" height="300" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/5/b/5b402f757d9f3ee96697e09f23a63b57e1e823e1_2_690x300.png, https://aws1.discourse-cdn.com/github/optimized/3X/5/b/5b402f757d9f3ee96697e09f23a63b57e1e823e1_2_1035x450.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/5/b/5b402f757d9f3ee96697e09f23a63b57e1e823e1.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/5/b/5b402f757d9f3ee96697e09f23a63b57e1e823e1_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">grafik</span><span class="informations">1274×554 35.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>The way I fix this is to fetch from upstream in <a href="http://github.com" rel="noopener nofollow ugc">github.com</a> and then sync using GitHub desktop. Is there a way to check on this and do this from the app?</p>

**Comment by steveward** [comment, marked as answer]
<p>:wave: <a class="mention" href="/danielniccoli">@danielniccoli</a> – in GitHub Desktop the <code>upstream</code> branches are available from the branches list, allowing you to merge changes from upstream and keep your fork in sync. You can go to <code>History</code> and use the <code>Select Branch to Compare</code> option at the top of the commit list to select a branch to compare to your current branch. I hope that helps!</p>

**Comment by danielniccoli** [comment]
<p>Hi, yeah that works. Even thought it is not really comfortable. I feel it would be better, if the app notifies me that changes happened on upstream, and ask me whether I like to sync them.</p>

**Comment by dstorozhuk** [comment]
Yes, that far a way from good UI to manually sync from upstream. Why that option (sinc from upsteram) available in WEB UI and issued on Desktop App? It is confusing.

**Comment by nuri-yoo** [comment]
If you want to check sync status across all your forks at once from the terminal (not just one at a time), I built [branch-watch](https://github.com/nuri-yoo/branch-watch) for this. `bw forks` shows every fork with ahead/behind commit counts vs. upstream in one view.
